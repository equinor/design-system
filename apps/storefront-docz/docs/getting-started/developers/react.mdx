---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
