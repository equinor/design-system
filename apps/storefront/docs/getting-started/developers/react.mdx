---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
