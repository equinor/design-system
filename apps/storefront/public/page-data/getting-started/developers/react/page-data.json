{"componentChunkName":"component---docs-getting-started-developers-react-mdx","path":"/getting-started/developers/react/","result":{"pageContext":{"frontmatter":{"title":"Developers","metaTitle":"Developers react","tabs":["Introduction","CSS framework","React","Figma"],"mode":"draft","route":"/getting-started/developers/react/"},"entry":{"id":"6f9784da00517d57e4627269567073e6","filepath":"docs/getting-started/developers/react.mdx","fullpath":"/Users/vnys/Kode/storefront-replace/apps/storefront/docs/getting-started/developers/react.mdx","route":"/getting-started/developers/react/","slug":"docs-getting-started-developers-react","name":"React","menu":"","headings":[]}}}}