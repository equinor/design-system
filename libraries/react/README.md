@eds-ui/react
===============

1. `npm i -g pnpm`
1. `pnpm install`
1. `pnpm recursive install`

## Run storybook

`pnpm start`

## Build modules

`pnpm recursive run build`
