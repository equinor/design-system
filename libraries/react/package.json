{
  "name": "@eds-ui/react",
  "private": true,
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "AGPL",
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@storybook/react": "^5.0.11",
    "babel-plugin-styled-components": "^1.10.0",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.12.1",
    "jest": "^24.8.0",
    "jest-styled-components": "^6.3.1",
    "rollup": "^1.11.3",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-node-resolve": "^4.2.3",
    "styled-components": "^4.2.0"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "styled-components": "^4.2.0",
    "prop-types": "^15.7.2"
  },
  "jest": {
    "verbose": true,
    "setupFilesAfterEnv": [
      "<rootDir>/testSetup.js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "transformIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "haste": {
      "providesModuleNodeModules": [
        ".*"
      ]
    },
    "roots": [
      "packages/node_modules"
    ]
  }
}
