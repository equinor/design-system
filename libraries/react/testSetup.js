import React from 'react'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })
