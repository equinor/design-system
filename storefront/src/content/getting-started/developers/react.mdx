---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
mode: draft
---


:warning: Information coming soon
