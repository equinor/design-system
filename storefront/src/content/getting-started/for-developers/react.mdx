---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
---

Developers React
