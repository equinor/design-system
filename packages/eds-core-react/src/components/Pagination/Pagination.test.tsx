/* eslint-disable no-undef */
import { render, cleanup, screen, within } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { Pagination } from '.'

const StyledPagination = styled(Pagination)`
  position: absolute;
`

afterEach(cleanup)

describe('Pagination', () => {
  it('Matches snapshot', () => {
    const { asFragment } = render(<Pagination totalItems={10} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(<Pagination totalItems={10} />)
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Should pass a11y test with aria label', async () => {
    const { container } = render(
      <Pagination totalItems={10} aria-label="description" />,
    )
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Should pass a11y test with default page', async () => {
    const { container } = render(<Pagination totalItems={10} defaultPage={3} />)
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Should pass a11y test with number of pages', async () => {
    const { container } = render(
      <Pagination totalItems={10} itemsPerPage={1} />,
    )
    expect(await axe(container)).toHaveNoViolations()
  })
  it('can extend the css for the component', () => {
    render(<StyledPagination totalItems={10} />)
    const pagination = screen.getByRole('navigation')
    expect(pagination).toHaveStyleRule('position', 'absolute')
  })
  it('has aria label', () => {
    render(<Pagination totalItems={10} />)
    const pagination = screen.getByRole('navigation')
    expect(pagination).toHaveAttribute('aria-label', 'pagination')
  })
  it('has aria label on pages', () => {
    render(<Pagination totalItems={100} defaultPage={3} />)
    expect(
      within(screen.getAllByRole('listitem')[0]).getByRole('button'),
    ).toHaveAttribute('aria-label', 'Go to previous page')
    expect(
      within(screen.getAllByRole('listitem')[1]).getByRole('button'),
    ).toHaveAttribute('aria-label', 'Go to page 1')
    expect(
      within(screen.getAllByRole('listitem')[3]).getByRole('button'),
    ).toHaveAttribute('aria-label', 'Current page, page 3')
  })
  it('should render correct number of pages', () => {
    render(<Pagination totalItems={4} itemsPerPage={1} />)
    expect(screen.getAllByRole('listitem')).toHaveLength(6) // totalItems + 2: the < + > buttons)
  })
  it('should render ellipsis if number of pages is larger than 7', () => {
    render(<Pagination totalItems={8} itemsPerPage={1} />)
    expect(screen.getByLabelText('Ellipsis of pages')).toBeDefined()
  })
  it('should render two ellipsises if number of pages is larger than 9 and default page is 5', () => {
    render(<Pagination totalItems={9} itemsPerPage={1} defaultPage={5} />)

    expect(screen.getAllByLabelText('Ellipsis of pages')[0]).toBeDefined()
    expect(screen.getAllByLabelText('Ellipsis of pages')[1]).toBeDefined()
  })
  it('should render next and previous buttons', () => {
    render(<Pagination totalItems={3} itemsPerPage={1} />)
    expect(screen.getAllByRole('button')[0]).toHaveAttribute(
      'aria-label',
      'Go to previous page',
    )
    expect(screen.getAllByRole('button')[4]).toHaveAttribute(
      'aria-label',
      'Go to next page',
    )
  })
})
