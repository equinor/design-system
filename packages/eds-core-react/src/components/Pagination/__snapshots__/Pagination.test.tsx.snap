// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Matches snapshot 1`] = `
<DocumentFragment>
  .c4 {
  display: grid;
  grid-gap: var(--eds_button__gap,8px);
  grid-auto-flow: column;
  align-items: center;
  height: 100%;
  justify-content: center;
}

.c4>:is(svg,img) {
  margin-top: var(--eds_button__icon__margin_y,0);
  margin-bottom: var(--eds_button__icon__margin_y,0);
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  text-decoration: none;
  position: relative;
  cursor: pointer;
  display: inline-block;
  background: transparent;
  height: 40px;
  width: 40px;
  padding-left: 0;
  padding-top: var(--eds_button__padding_y, 0);
  padding-right: 0;
  padding-bottom: var(--eds_button__padding_y, 0);
  border: 0px solid transparent;
  border-radius: 50%;
  margin: 0;
  color: var(--eds_interactive_primary__resting, rgba(0, 112, 121, 1));
  font-family: Equinor;
  font-size: var(--eds_button__font_size, 0.875rem);
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
}

.c3 svg {
  justify-self: center;
}

.c3::after {
  position: absolute;
  top: -4px;
  left: -4px;
  width: 48px;
  height: 48px;
  content: '';
}

.c3:focus {
  outline: none;
}

.c3[data-focus-visible-added]:focus {
  outline: 2px dashed rgba(0, 112, 121, 1);
  outline-offset: -2px;
}

.c3:focus-visible {
  outline: 2px dashed rgba(0, 112, 121, 1);
  outline-offset: -2px;
}

.c3::-moz-focus-inner {
  border: 0;
}

.c3:disabled,
.c3[aria-disabled='true'] {
  cursor: not-allowed;
  background: transparent;
  border: 1px solid transparent;
  margin: 0;
  color: var(--eds_interactive__disabled__text, rgba(190, 190, 190, 1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
}

.c5 {
  transform: none;
}

.c0 {
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: nowrap;
  margin-left: 0;
}

.c1 {
  list-style: none;
  margin: 0;
  padding: 0;
  display: grid;
  grid-gap: 8px;
  grid-auto-flow: column;
}

.c2 {
  display: inline-grid;
}

@media (hover:hover) and (pointer:fine) {
  .c3:hover {
    background: var(--eds_interactive_primary__hover_alt, rgba(222, 237, 238, 1));
    border: 0px solid transparent;
    border-radius: 50%;
  }
}

@media (hover:hover) and (pointer:fine) {
  .c3:disabled:hover,
  .c3[aria-disabled='true']:hover {
    background: transparent;
  }
}

<nav
    aria-label="pagination"
    class="c0"
  >
    <ol
      class="c1"
      style="grid-template-columns: repeat(3, 48px);"
    >
      <li
        class="c2"
      >
        <button
          aria-label="Go to previous page"
          class="c3"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="c4"
          >
            <svg
              aria-hidden="true"
              class="c5"
              fill="currentColor"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                clip-rule="evenodd"
                d="M15.705 16.59 11.125 12l4.58-4.59L14.295 6l-6 6 6 6z"
                data-testid="eds-icon-path"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </li>
      <li
        class="c2"
      >
        <button
          aria-current="1"
          aria-label="Current page, page 1"
          class="c3"
          tabindex="0"
          type="button"
        >
          <span
            class="c4"
          >
            1
          </span>
        </button>
      </li>
      <li
        class="c2"
      >
        <button
          aria-label="Go to next page"
          class="c3"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="c4"
          >
            <svg
              aria-hidden="true"
              class="c5"
              fill="currentColor"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                clip-rule="evenodd"
                d="m8.295 16.59 4.58-4.59-4.58-4.59L9.705 6l6 6-6 6z"
                data-testid="eds-icon-path"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
